hw/chip/mimxrt10xx/mimxrt101x/pinning_options.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020, Dave Marples
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+#ifndef _PINNING_OPTIONS_H_
+#define _PINNING_OPTIONS_H_
+
+// UART Options =====================================
+
+#if defined(PINNING_UART_LPUART1_GPIO_09_10)
+
+#define UART_PORT       LPUART1
+#define PIN_UART_RX     IOMUXC_GPIO_09_LPUART1_RXD
+#define PIN_UART_RX_MUX 0U
+#define PIN_UART_RX_CFG 0x10B0u
+#define PIN_UART_TX     IOMUXC_GPIO_10_LPUART1_TXD
+#define PIN_UART_TX_MUX 0U
+#define PIN_UART_TX_CFG 0x10B0u
+
+#else
+
+#error No UART pinning set
+
+#endif
+
+// LED Options ======================================
+
+#if defined(PINNING_LED_GPIO_11_H)
+
+#define PIN_LED         IOMUXC_GPIO_11_GPIOMUX_IO11
+#define PIN_LED_MUX     0U
+#define PIN_LED_CFG     0x10B0U
+#define LED_GPIO_PORT   GPIO1
+#define LED_GPIO_PIN    11
+#define LED_STATE_ON    1
+
+#else
+
+#error No LED pinning set
+
+#endif
+
+// QSPI Options =====================================
+
+#if defined(PINNING_QSPI_FLEXSPIA)
```

Comment:
This should stay here cause there are only 2 options, although the _CFG will always be same so could go directly into the .c file.

Reply:
Agreed, but lets not move the CFG into the c as magic numbers ... I would prefer to keep the whole CFG/MUX/Settings in one place rather than sending people scurrying through multiple files if they ever need changing (shades of MISRA rules)